api/v3/utils.php

```diff
@@ -1931,17 +1998,23 @@ function _civicrm_api3_api_match_pseudoconstant(&$params, $entity, $fieldName, $
     }
     $options = civicrm_api($entity, 'getoptions', array('version' => 3, 'field' => $fieldInfo['name'], 'context' => 'validate'));
     $options = CRM_Utils_Array::value('values', $options, array());
+
+    if (count($options) == 0 && isset($pseudoconstant['table'])) {
+        $pseudoParams = $pseudoconstant;
+        unset($pseudoParams['table']);
+        $options = CRM_Core_PseudoConstant::get(CRM_Core_DAO_AllCoreTables::getClassForTable($pseudoconstant['table']), $fieldName, $pseudoParams);
```

Comment:
I have no idea. @monishdeb could you please add code comments here so we can understand why it is necessary.